AGENTS.md

```diff
@@ -0,0 +1,202 @@
+
+- Operator reconciliation logic for managing monitoring components
+- Jsonnet-based manifest generation system for Kubernetes resources
+- Configuration API for platform and user-workload monitoring
+- Task-based architecture for component lifecycle management
+- Integration with OpenShift FeatureGates and cluster configuration
+
+## Key Architecture Components
+
+### Task-Based Reconciliation System
+The operator uses a task-based architecture (`pkg/tasks/`) where each monitoring component is managed by a dedicated task. Tasks are organized into groups that execute sequentially, with tasks within each group running concurrently. Key tasks include:
+- `PrometheusOperatorTask` - Manages prometheus-operator deployment
+- `PrometheusTask` - Manages platform Prometheus instance
+- `AlertmanagerTask` - Manages Alertmanager cluster
+- `NodeExporterTask` - Manages node-exporter DaemonSet
+- `KubeStateMetricsTask` - Manages kube-state-metrics
+- `ThanosQuerierTask` - Manages Thanos Querier for multi-cluster queries
+- User Workload Monitoring tasks (when enabled)
+
+### Manifest Generation System
+CMO uses Jsonnet (`jsonnet/`) to generate Kubernetes manifests:
+- `jsonnet/main.jsonnet` - Entry point for manifest generation
+- `jsonnet/components/*.libsonnet` - Component-specific manifest definitions
+- `jsonnet/utils/*.libsonnet` - Utility functions for manifest manipulation
+- `assets/` - Generated YAML manifests embedded in the operator binary
+- Vendored upstream jsonnet from kube-prometheus and other projects
+
+### Package Structure
+- `pkg/manifests/` - Manifest loading, parsing, and configuration API types
+- `pkg/tasks/` - Task implementations for each monitoring component
+- `pkg/operator/` - Main operator reconciliation loop and controller logic
+- `pkg/client/` - Kubernetes client wrappers and utilities
+- `pkg/alert/` - Alert configuration and validation
+- `pkg/metrics/` - Operator metrics and instrumentation
+- `pkg/promqlgen/` - PromQL query generation utilities
+
+### Configuration Management
+Configuration is managed through two ConfigMaps:
+- `cluster-monitoring-config` (in `openshift-monitoring`) - Platform monitoring configuration
+- `user-workload-monitoring-config` (in `openshift-user-workload-monitoring`) - User workload monitoring configuration
+
+Configuration types are defined in `pkg/manifests/types.go` with extensive validation.
+
+## Common Development Commands
+
+### Building
+```bash
+make build              # Build operator and tests-ext binaries
+make operator           # Build only the operator binary
+make operator-no-deps   # Build operator without fetching dependencies
+```
+
+### Jsonnet Manifest Generation
+```bash
+make generate           # Generate all manifests, docs, and test metadata
+make build-jsonnet      # Generate YAML manifests from jsonnet
+make check-assets       # Verify generated assets are up to date
+```
+
+### Local Development
+```bash
+export KUBECONFIG=/path/to/kubeconfig
```

Comment:
worth noting that this needs an OpenShift cluster?

Reply:
I can remove this, but I still do think that this may be needed for an agent.